### ISSUE TITLE
Running tests on Android

### ISSUE THREAD

**Issue by amatkivskiy** [jayjaylab-benchmark/retrofitvsokhttp#1]
Hi @jayjaykim 
Thanks for benchmarks.
As I see you used [JMH tool](http://openjdk.java.net/projects/code-tools/jmh/) for benchmarking. You run tests against **JDK 1.8.0_45** but do you think that tests results might be different when they are run on ART ? If yes, do you any way to run JMH on Android runtime?

**Comment by jayjaykim** [comment]
You are right. I run the benchmark on Hotspot JVM not in ART or dalvik. Because JMH doesn't support the benchmark run on ART or dalvik which is so sad. I couldn't find any benchmark tools  which can support on Android runtime yet.
In my opinion, the result numbers can be different if it can run on android runtime, but I guess the pattern of the result will not change.

**Comment by amatkivskiy** [comment]
@jayjaykim I found [Caliper](https://github.com/google/caliper) benchmarks can be run through [Vogar](https://code.google.com/archive/p/vogar/). Unfortunately I have not succeed in configuring this set of tools (Vogar is really old). 
In such situations I am afraid that ART will make some optimizations which will influence the performance tests on Android.

**Comment by jayjaykim** [comment]
Definitely, ART will influence everythings to be estimated in a benchmark. But can you imagine that java developers write optimized codes for a specific JVM?
Suppose, there are A, B, C algorithms to be estimated in a benchmark. A shows superior than B, C on hotspot JVM. Is it likely that B or C is superior than A when it comes to ART or dalvik or any other JVMs? Well, I'm negative on that. But it's my guess, we as engineers need facts and results of experiments...

**Comment by amatkivskiy** [comment]
@jayjaykim Yeah, you are completely right. Do you know any tool to measure memory pressure of a method like we are doing for performance testing?

**Comment by jayjaykim** [comment]
do you mean something like MAT?

**Comment by amatkivskiy** [comment]
@jayjaykim yes.

**Comment by jayjaykim** [comment]
hmm.. What I know is MAT, memory monitor function in Android studio, memory dump.... Sorry those may be what you already know..
